include/unifex/let_error.hpp

```diff
@@ -201,167 +206,202 @@ class _frcvr<Source, Error, Receiver>::type {
-  template(typename Source, typename Error)
-    (requires tag_invocable<_fn, Source, Error> AND
-        sender<Source> AND
-        callable<remove_cvref_t<Error>> AND
-        sender<callable_result_t<remove_cvref_t<Error>>>)
-  auto operator()(Source&& source, Error&& error) const
-      noexcept(is_nothrow_tag_invocable_v<_fn, Source, Error>)
-      -> tag_invoke_result_t<_fn, Source, Error> {
-    return tag_invoke(*this, (Source&&)source, (Error&&)error);
+  template(typename Source, typename Func)
+    (requires tag_invocable<_fn, Source, Func> AND
+        sender<Source>)
+        // callable<remove_cvref_t<Error>> AND
+        // sender<callable_result_t<remove_cvref_t<Error>>>)
+  auto operator()(Source&& source, Func&& func) const
+      noexcept(is_nothrow_tag_invocable_v<_fn, Source, Func>)
+      -> tag_invoke_result_t<_fn, Source, Func> {
+    return tag_invoke(*this, (Source&&)source, (Func&&)func);
   }
 
-  template(typename Source, typename Error)
-    (requires (!tag_invocable<_fn, Source, Error>) AND
+  template(typename Source, typename Func)
+    (requires (!tag_invocable<_fn, Source, Func>) AND
         constructible_from<remove_cvref_t<Source>, Source> AND
-        constructible_from<remove_cvref_t<Error>, Error> AND
-        callable<remove_cvref_t<Error>> AND
-        sender<callable_result_t<remove_cvref_t<Error>>>)
-  auto operator()(Source&& source, Error&& error) const
+        constructible_from<remove_cvref_t<Func>, Func>)
+        // callable<remove_cvref_t<Error>> AND
+        // sender<callable_result_t<remove_cvref_t<Error>>>)
+  auto operator()(Source&& source, Func&& func) const
       noexcept(std::is_nothrow_constructible_v<
-                   _sender<remove_cvref_t<Source>, remove_cvref_t<Error>>,
-                   Source, 
-                   Error>)
-      -> _sender<remove_cvref_t<Source>, remove_cvref_t<Error>> {
-    return _sender<remove_cvref_t<Source>, remove_cvref_t<Error>>{
-        (Source&&)source, (Error&&)error};
+                   _sender<remove_cvref_t<Source>, remove_cvref_t<Func>>,
+                   Source,
+                   Func>)
+      -> _sender<remove_cvref_t<Source>, remove_cvref_t<Func>> {
+    return _sender<remove_cvref_t<Source>, remove_cvref_t<Func>>{
+        (Source&&)source, (Func&&)func};
   }
-  template(typename Error)
-      (requires callable<remove_cvref_t<Error>> AND
-        sender<callable_result_t<remove_cvref_t<Error>>>)
-  constexpr auto operator()(Error&& error) const
+  template<typename Func>
+      // (requires callable<remove_cvref_t<Error>> AND
+      //   sender<callable_result_t<remove_cvref_t<Error>>>)
+  constexpr auto operator()(Func&& func) const
       noexcept(is_nothrow_callable_v<
-        tag_t<bind_back>, _fn, Error>)
-      -> bind_back_result_t<_fn, Error> {
-    return bind_back(*this, (Error&&)error);
+        tag_t<bind_back>, _fn, Func>)
+      -> bind_back_result_t<_fn, Func> {
+    return bind_back(*this, (Func&&)func);
   }
```

Comment:
IMO, It is a good idea to remove the commented out constraints for these operator() overloads. They are misleading since they refer to Error instead of Func and they expect the func to be callable with no args, which is no longer supported.

Reply:
We could probably specify that `Func` is a callable for all possible tuples of `Sender`'s `error_types` but that won't look pretty, so I just deleted these comments indeed.